master/buildbot/data/base.py

```diff
@@ -72,7 +72,7 @@ def compileEventPathPatterns(self):
             pathPatterns[i] = pp
         self.eventPaths = pathPatterns
 
-    @functools.lru_cache(1)
+    @functools.lru_cache(1)  # noqa: B019
```

Comment:
I looked at the cases and given there is not an unbound number of instances, this memory leak is fine. But in general people should be aware of this issue and review if their usage is acceptable.